mava/utils/jax.py

```diff
@@ -42,3 +47,18 @@ def merge_leading_dims(x: chex.Array, num_dims: chex.Numeric) -> chex.Array:
 
     new_shape = (np.prod(x.shape[:num_dims]),) + x.shape[num_dims:]
     return x.reshape(new_shape)
+
+
+def unreplicate_learner_state(
+    learner_state: Union[LearnerState, RNNLearnerState]
+) -> Union[LearnerState, RNNLearnerState]:
+    """Unreplicates a learner state.
+
+    This function takes a learner state, and removes the axes associated with device replication
+    and the `update batch size`. The unreplication process is essential to store the learner state
+    in a checkpoint.
+
+    Note:
+        The function internally uses `jax_utils.unreplicate` twice to remove the necessary axes.
+    """
+    return jax_utils.unreplicate(jax_utils.unreplicate(learner_state))  # type: ignore
```

Comment:
Perhaps we can replace this with a single tree map? Not a bottleneck in speed, but just a little cleaner :)
```suggestion
    return jax.tree_map(lambda x: x[0][0], learner_state)
```

Confirmed that these are equivalent:
```
chex.assert_trees_all_equal(
    jax.tree_map(lambda x: x[0][0], learner_state),
    jax_utils.unreplicate(jax_utils.unreplicate(learner_state)),
)
```

Reply:
Actually, I have another recommendation—give me a sec :)

Reply:
see below :)

Reply:
much cleaner thanks for the suggestion